### PR TITLE
Closes #8 Job show dynamics and tests

### LINKED ISSUE #8: Guest Can View Job Show
As a guest,
When I visit "/users/#{user_name}/jobs/#{job.id}",
And the bidding for the job is open,
I do not see the "Place a Bid" accordion,
And I see the "Log in to Place a Bid" button in its place,
And I do see info for the job:
- A link back to the category show
- The name of the job
- The full description of the job
- Estimated duration tag
- "Must be completed by" date
- "Bidding ends" date
- Bid Range
- Number of Total Bids
- Location of the job
- Lister's Information showing:
  - Poster's name as link to profile
  - Rating
  - Poster's image
  - Number of active jobs
  - Number of completed jobs
  - "Active Since" date

### LINKED ISSUE #40: Contractor Job Show
As a contractor,
When I visit the job show page,
And the job's bidding is open,
Then I see:
A link back to the category show
The name of the job
The full description of the job
Estimated duration tag
"Must be completed by" date
"Bidding ends" date
Total # of bids
Bid Range
Location of the job
Lister's Information showing:
Name as link to user show page
Rating
User's Picture
Number of active jobs
Number of completed jobs
"Active Since" date
"Place a Bid" accordion which has:
- My name
- Price box
- Duration estimate box
- Description box
- Submit bid button

### LINKED ISSUE #41: Lister Job Show
As a lister,
When I visit "/users/#{user_name}/jobs/#{job.id}" for a job I listed,
And the bidding for that job is open,
I do not see the "Place a Bid" accordion,
And I do not see the "Log in to Place a Bid" button in its place,
And I do see "Current Bids" table, containing columns:
- Bidder Name
- Price
- Duration Estimate
- Comment
- Action

And I do see info for the job:
A link back to the category show
The name of the job
The full description of the job
Estimated duration tag
"Must be completed by" date
"Bidding ends" date
Total # of Bids
Bid Range
Location of the job
Lister's Information showing:
Poster's name as link to profile
Poster's image
Rating
Number of active jobs
Number of completed jobs
"Active Since" date

### LINKED ISSUE #85: Lister Accepts Bid From Job Show Page
As a logged in job lister,
With a job that has pending bids and the job status is "Bidding Closed",
When I visit the job show page for that job,
I see a table of the current bids,
And I see that each bid has a link to "Accept" it,
And when I click on the first "Accept" link,
That bid's status changes to accepted,
And all other bid's statuses change to rejected,
And I see that the bid I accepted now says "Accepted" in the table of current bids,
And I see that all other bids say "Rejected" in the table of current bids,
And there is a banner on the job show page stating "Job Is In Progress"

### PR DESCRIPTION
**Pull request by ToniRib**
Also closes #41 
closes #85 
closes #40 

This long string of commits sets up the Job Show page to be dynamic and includes all of the conditions for various banners about the job status based on the job status. Additionally, it includes functionality for contractors and listers to place a bid (though listers cannot if they are the ones who listed the job). Also forces guests to log in if they want to place a bid.

### CODE REVIEW

File: app/assets/stylesheets/_custom.scss [hunk 1]
```diff
@@ -220,6 +254,33 @@ button.stripe-button-el span {
...
+  margin: 5px auto;
+  width: 168px;
+
+  &:hover {
+    background-color: $dark-red;
+  }
+
+}
+
+a.btn.btn-delete {
```
Comment (ToniRib) [inline app/assets/stylesheets/_custom.scss:279]: Ignoring this css hound comment

File: app/assets/stylesheets/_custom.scss [hunk 2]
```diff
@@ -240,6 +301,32 @@ h3#ui-id-1 {
     .arrow-icon {
       color: $base-green-dark;
     }
+
+  }
+
+}
+
+h3#ui-id-1.my-bid {
```
Comment (ToniRib) [inline app/assets/stylesheets/_custom.scss:309]: Ignoring this, this was the only way I got it to work.

File: app/assets/stylesheets/_media_queries.scss [hunk 3]
```diff
@@ -129,6 +135,15 @@
     width: 250px;
   }
 
+  .btn-delete,
+  .update-bid-btn {
+    margin: 10px auto;
+  }
+
+  h3#ui-id-1.my-bid {
```
Comment (ToniRib) [inline app/assets/stylesheets/_media_queries.scss:143]: Ignoring this, I had to do this to get it to work with the accordion.

File: test/integration/user_places_bid_test.rb [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  end
+
+  test "contractor cannot see other contractor's bids" do
+    contractor_1 = create(:contractor)
+    contractor_2 = create(:contractor)
+    job = create(:job)
+    lister = job.lister
+    bid = create(:bid, job_id: job.id, user_id: contractor_1.id)
+
+    ApplicationController.any_instance.stubs(:current_user).returns(contractor_2)
```
Comment (ToniRib) [inline test/integration/user_places_bid_test.rb:92]: Not much I can do about this.

File: app/controllers/user/bids_controller.rb [hunk 5]
```diff
@@ -1,4 +1,60 @@
...
+      redirect_to request.referrer
+    end
+  end
+
+  def update
+    if params[:status]
+      update_bid_and_job_status
+      redirect_to request.referrer
+    else
+      if @bid.update_attributes(bid_params)
```
Comment (ToniRib) [inline app/controllers/user/bids_controller.rb:21]: Nope, I don't see why I would do this since it is the start of a separate if statement.

**Comment by ToniRib** [comment]
Ignoring the rest of the hound complaints since I know I am about to refactor a lot of this anyway.

File: app/presenters/job_presenter.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+class JobPresenter < SimpleDelegator
+  attr_reader :view, :model
+
+  def initialize(model, view)
+    @model = model
+    @view = view
+    super(@model)
+  end
+end
```
Comment (jwperry) [inline app/presenters/job_presenter.rb:9]: Nice work with the presenters!

File: app/views/user/jobs/_place_a_bid_accordion.html.erb [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+            <%= label_tag :days %>
+          </div>
+        </div>
+        <div class="row no-bottom-margin center-align valign-wrapper">
+          <div class="col s12 m8 valign">
+            <%= f.text_area :details, placeholder: "Leave a comment for the job lister",
+                                      class: "bid-comment",
+                                      required: true,
+                                      maxlength: 400,
+                                      minlength: 35 %>
```
Comment (jwperry) [inline app/views/user/jobs/_place_a_bid_accordion.html.erb:30]: These requirements and error pop-ups rock.

**Pull request merged**